static/app/views/performance/transactionSummary/transactionTags/tagsHeatMap.tsx

```diff
@@ -238,21 +239,158 @@ const TagsHeatMap = (
     } as any); // TODO(k-fish): Fix heatmap data typing
   }
 
-  const onOpenMenu = () => {
-    setIsMenuOpen(true);
-  };
+  const onChartClick = bucket => {
+    const htmlEvent = bucket.event.event;
+    // Make a copy of the dims because echarts can remove elements after this click happens.
+    // TODO(k-fish): Look at improving this to respond properly to resize events.
+    const virtualRef = new VirtualReference(htmlEvent.target);
+    setChartElement(virtualRef);
+
+    const newTransactionEventView = eventView.clone();
+
+    newTransactionEventView.fields = [{field: aggregateColumn}];
+    const [_, tagValue] = bucket.value;
+
+    if (histogramBucketInfo && histogramData) {
+      const row = histogramData[bucket.dataIndex];
+      const currentBucketStart = parseInt(
+        `${row[histogramBucketInfo.histogramField]}`,
+        10
+      );
+      const currentBucketEnd = currentBucketStart + histogramBucketInfo.bucketSize;
+
+      newTransactionEventView.additionalConditions.setFilterValues(aggregateColumn, [
+        `>=${currentBucketStart}`,
+        `<${currentBucketEnd}`,
+      ]);
+    }
 
-  const onCloseMenu = () => {
-    setIsMenuOpen(false);
+    if (tagKey) {
+      newTransactionEventView.additionalConditions.setFilterValues(tagKey, [tagValue]);
+    }
+
+    setTransactionEventView(newTransactionEventView);
+    trackTagPageInteraction(organization);
+
+    if (!overlayState.isOpen) {
+      overlayState.open();
+    }
   };
 
-  const shouldIgnoreMenuClose = e => {
-    if (chartRef.current?.getEchartsInstance().getDom().contains(e.target)) {
+  const overlayState = useOverlayTriggerState({});
+  const {overlayProps} = useOverlay(
+    {
+      isOpen: overlayState.isOpen,
+      onClose: overlayState.close,
+      isDismissable: true,
+      shouldCloseOnBlur: true,
       // Ignore the menu being closed if the echart is being clicked.
-      return true;
+      shouldCloseOnInteractOutside: el =>
+        !chartRef.current?.getEchartsInstance().getDom().contains(el),
+    },
+    {current: overlayElement ?? null}
+  );
+
+  const {styles: popperStyles, state: popperState} = usePopper(
+    chartElement,
+    overlayElement,
+    {
+      placement: 'bottom',
+      strategy: 'absolute',
+      modifiers: [
+        {name: 'computeStyles', options: {gpuAcceleration: false}},
+        {name: 'offset', options: {offset: [0, 8]}},
+        {name: 'arrow', options: {element: overlayArrowElement, padding: 4}},
+        {name: 'preventOverflow', enabled: true, options: {padding: 12, altAxis: true}},
+      ],
     }
-    return false;
-  };
+  );
+
+  const theme = useTheme();
+  const portaledContent =
+    !chartElement || !overlayState.isOpen ? null : (
+      <PositionWrapper
+        ref={setOverlayElement}
+        zIndex={theme.zIndex.dropdown}
+        style={popperStyles.popper}
+        {...(overlayProps as HTMLMotionProps<'div'>)}
```

Comment:
`overlayProps` has type `HTMLAttributes<HTMLDivElement>`, which doesn't exactly match with `PositionWrapper`'s expectation of `HTMLMotionProps<'div'>`:

<img width="1020" alt="image" src="https://user-images.githubusercontent.com/44172267/177652787-2665189c-264a-4457-b6cd-b5515c155f2e.png">

This happens because framer motion components (`motion.div`) have custom event handlers that overwrite default DOM handlers like `onAnimationStart`.

One way to fix this: in `Overlay.tsx`, instead of extending [`OverlayProps`](https://github.com/getsentry/sentry/blob/master/static/app/components/overlay.tsx#L13) and [`PositionWrapperProps`](https://github.com/getsentry/sentry/blob/master/static/app/components/overlay.tsx#L136) from `HTMLMotionProps<'div'>`, we can extend them from `HTMLAttributes<HTMLDivElement>`. This would be fine if we're going to define overlay animations internally inside `Overlay`, i.e. when using `Overlay`/`PositionWrapper` we'd only pass in a string/boolean specifying the animation type/whether we want an animation (`animationType`/`animated`) and `Overlay` will parse that and turn it into `motion.div` props.